lte/cloud/go/services/subscriberdb/digest_test.go

```diff
@@ -251,3 +252,193 @@ func TestGetDigestApnResourceAssocs(t *testing.T) {
+	var writes []configurator.EntityWriteOperation
+	writes = append(writes, configurator.NetworkEntity{
+		NetworkID: "n1",
+		Type:      lte.APNResourceEntityType,
+		Key:       "resource",
+		Config: &lte_models.ApnResource{
+			ApnName:    "apn",
+			GatewayIP:  "172.16.254.1",
+			GatewayMac: "00:0a:95:9d:68:16",
+			ID:         "resource",
+			VlanID:     42,
+		},
+		Associations: storage.TKs{{Type: lte.APNEntityType, Key: "apn"}},
+	})
+	write := configurator.EntityUpdateCriteria{
+		Type:              lte.CellularGatewayEntityType,
+		Key:               gw.Key,
+		AssociationsToAdd: storage.TKs{{Type: lte.APNResourceEntityType, Key: "resource"}},
+	}
+	writes = append(writes, write)
+	err = configurator.WriteEntities("n1", writes, serdes.Entity)
+	assert.NoError(t, err)
+
+	perSubDigests, err = subscriberdb.GetPerSubDigests("n1")
+	assert.NoError(t, err)
+	assert.Equal(t, "0001", perSubDigests[0].Sid.Id)
+	assert.NotEqual(t, digestSub1, perSubDigests[0].Digest.Md5Base64Digest)
+	assert.Equal(t, "0003", perSubDigests[1].Sid.Id)
+	assert.NotEqual(t, digestSub3, perSubDigests[1].Digest.Md5Base64Digest)
+}
+
+func TestGetPerSubDigestsDiff(t *testing.T) {
+	t.Run("both empty", func(t *testing.T) {
+		all, tracked := []*lte_protos.SubscriberDigestByID{}, []*lte_protos.SubscriberDigestByID{}
+		toRenew, deleted := subscriberdb.GetPerSubDigestsDiff(all, tracked)
+		assert.Equal(t, map[string]string{}, toRenew)
+		assert.Equal(t, []string{}, deleted)
+	})
+	t.Run("all empty", func(t *testing.T) {
+		all := []*lte_protos.SubscriberDigestByID{}
+		tracked := []*lte_protos.SubscriberDigestByID{
+			{Sid: &lte_protos.SubscriberID{Id: "00000"}, Digest: &lte_protos.Digest{Md5Base64Digest: "apple"}},
+			{Sid: &lte_protos.SubscriberID{Id: "00001"}, Digest: &lte_protos.Digest{Md5Base64Digest: "banana"}},
+			{Sid: &lte_protos.SubscriberID{Id: "00002"}, Digest: &lte_protos.Digest{Md5Base64Digest: "cherry"}},
+		}
+		toRenew, deleted := subscriberdb.GetPerSubDigestsDiff(all, tracked)
+
+		assert.Equal(t, map[string]string{}, toRenew)
+		assert.Equal(t, []string{"00000", "00001", "00002"}, deleted)
+	})
+	t.Run("tracked empty", func(t *testing.T) {
+		all := []*lte_protos.SubscriberDigestByID{
+			{Sid: &lte_protos.SubscriberID{Id: "00000"}, Digest: &lte_protos.Digest{Md5Base64Digest: "apple"}},
+			{Sid: &lte_protos.SubscriberID{Id: "00001"}, Digest: &lte_protos.Digest{Md5Base64Digest: "banana"}},
+			{Sid: &lte_protos.SubscriberID{Id: "00002"}, Digest: &lte_protos.Digest{Md5Base64Digest: "cherry"}},
+		}
+		tracked := []*lte_protos.SubscriberDigestByID{}
+		toRenew, deleted := subscriberdb.GetPerSubDigestsDiff(all, tracked)
+
+		assert.Equal(t, map[string]string{
+			"00000": "apple",
+			"00001": "banana",
+			"00002": "cherry",
+		}, toRenew)
+		assert.Equal(t, []string{}, deleted)
+	})
+	t.Run("both not empty, case 1", func(t *testing.T) {
+		all := []*lte_protos.SubscriberDigestByID{
```

Comment:
Here and elsewhere: "all" and "tracked" don't connote the difference between the two. Which is which?

Maybe something like "prev" and "next" ?